fbpic/particles/deposition/cuda_methods.py

```diff
@@ -575,86 +572,81 @@ def deposit_chi_gpu_linear(x, y, z, w, q, M, inv_gamma,
         if i == 0:
             frequency_per_cell = np.int32(incl_offset)
 
-        for it in range(len(m_tuple)):
-
-            m = m_tuple[it]
-            chi_m = chi_tuple[it]
-
-            # Declare local field arrays
-            R_m_00 = 0. + 0.j
-            R_m_01 = 0. + 0.j
-            R_m_10 = 0. + 0.j
-            R_m_11 = 0. + 0.j
-
-            for j in range(frequency_per_cell):
-                # Get the particle index before the sorting
-                # --------------------------------------------
-                # (Since incl_offset is a cumulative sum of particle number,
-                # and since python index starts at 0, one has to add -1)
-                ptcl_idx = incl_offset-1-j
-
-                # Preliminary arrays for the cylindrical conversion
-                # --------------------------------------------
-                # Position
-                xj = x[ptcl_idx]
-                yj = y[ptcl_idx]
-                zj = z[ptcl_idx]
-                # Gamma factor
-                inv_gammaj = inv_gamma[ptcl_idx]
-                # Weights
-                wj = q**2 / M * inv_gammaj * w[ptcl_idx]
-
-                # Cylindrical conversion
-                rj = math.sqrt(xj**2 + yj**2)
-                # Avoid division by 0.
-                if (rj != 0.):
-                    invr = 1./rj
-                    cos = xj*invr  # Cosine
-                    sin = yj*invr  # Sine
-                else:
-                    cos = 1.
-                    sin = 0.
-                # Calculate azimuthal factor
-                exptheta_m = 1. + 0.j
-                for _ in range(m):
-                    exptheta_m *= (cos + 1.j*sin)
```

Comment:
Since `m` can be negative, the above needs to be corrected. (see the corresponding corrected line)